### PR TITLE
Add types & update variable names in cloudConsole.ts

### PR DESCRIPTION
**Pull request by wwlorey**
These are purely engineering changes. And I added a test case for cloud shell to the onenote which passes for these changes.

### CODE REVIEW

**Comment by wwlorey** [comment]
Related: https://github.com/microsoft/vscode-azure-account/issues/313

**Review by nturinski — APPROVED** [review]

File: src/cloudConsole/cloudConsole.ts [hunk 1]
```diff
@@ -334,33 +330,38 @@ export function createCloudConsole(api: AzureAccountExtensionApi, reporter: Tele
...
 					return undefined;
 				})))
 				.then(tenantDetails => tenantDetails.filter(details => details));
+
 			const pick = await window.showQuickPick<QuickPickItem & { session: AzureSession }>(fetchingDetails
 				.then(tenantDetails => tenantDetails.map(details => {
 					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-					const tenantDetails = details!.tenantDetails;
-					const defaultDomain = tenantDetails.verifiedDomains.find(domain => domain.default);
+					const tenantDetails: TenantDetails = details!.tenantDetails;
```
[the remark this answers is not in the document]
Comment (wwlorey) [reply src/cloudConsole/cloudConsole.ts:350]: This line above filters out the undefined values so we can assume `details` is non-null here:

https://github.com/microsoft/vscode-azure-account/blob/9d8d27489ca4bf1401cb7b7e38d6f49aeab1b1aa/src/cloudConsole/cloudConsole.ts#L345

[the remark this answers is not in the document]
Comment (nturinski) [reply src/cloudConsole/cloudConsole.ts:350]: If they cancel?

[the remark this answers is not in the document]
Comment (wwlorey) [reply src/cloudConsole/cloudConsole.ts:350]: If I understand it right: `fetchingDetails` resolves and gives `window.showQuickPick` an array of `details` objects (with no undefined values). If the user cancels out of the quickpick we catch the undefined `pick` value here:

https://github.com/microsoft/vscode-azure-account/blob/9d8d27489ca4bf1401cb7b7e38d6f49aeab1b1aa/src/cloudConsole/cloudConsole.ts#L362

[the remark this answers is not in the document]
Comment (nturinski) [reply src/cloudConsole/cloudConsole.ts:350]: Huh, interesting.  I'm not really sure what the order of operations is here if you have an await and a .then statement... 🤔 But I'll take your word for it 🤣

[the remark this answers is not in the document]
Comment (nturinski) [reply src/cloudConsole/cloudConsole.ts:350]: Okay, I see now.  I didn't realize the .then was actually for getting item[], at first review, it looked like it was a .then _after_ the quick pick was selected.  I get it now.

**Pull request merged**